relay-general/src/store/trimming.rs

```diff
@@ -773,6 +767,32 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_frameqty_equals_limit() {
+        fn create_frame(filename: &str) -> Annotated<Frame> {
+            Annotated::new(Frame {
+                filename: Annotated::new(filename.into()),
+                ..Default::default()
+            })
+        }
+
+        let mut frames = Annotated::new(vec![
+            create_frame("foo3.py"),
+            create_frame("foo4.py"),
+            create_frame("foo5.py"),
+        ]);
+
+        frames
+            .apply(|f, m| {
+                enforce_frame_hard_limit(f, m, 3);
+                Ok(())
+            })
+            .unwrap();
+
+        // original_length isn't set, when limit is equal to length, as no trimming took place.
+        assert!(frames.meta().original_length().is_none());
+    }
+
```

Comment:
Probably overkill, just checks that no trimming took place when limit is equal to length, unlike previous implementation